docs/api.md

```diff
@@ -780,6 +780,21 @@ PATCH /api/users/1
 <- {}
 ```
 
+<a name='check-username-available'></a>
+### Check if a username is available [GET /api/username-available/:username]
```

Comment:
Hey, we could maybe generalize this for other endpoints. `GET /api/{channels,users}/name-available/:name`

Reply:
Ideally, yeah, but it'd be best to figure out how to first prevent nedb/shortid from generating `name-available` as an ID!

Reply:
(Also `/api/roles/name-available` if we go with that system.)